assets/js/util.js

```diff
@@ -293,13 +300,82 @@ function callApy(options, endpoint) {
     var requestUrl = window.location.protocol + window.location.hostname +
         window.location.pathname + '?' + $.param(requestOptions.data);
 
+    apyRequestStartTime = Date.now();
     if(requestUrl.length > THRESHOLD_REQUEST_URL_LENGTH) {
         requestOptions.type = 'POST';
         return $.ajax(requestOptions);
     }
     return $.jsonp(requestOptions);
 }
 
+function checkServiceLoadTimes(requestDuration) {
+    var lastNRequestsDuration = [];
+    var storedLastNRequestsDuration = [];
+    var demoThreshold = 10000; // for demo purposes only
+    try {
+        window.sessionStorage;
```

Comment:
> From this I assumed that session storage persisted only through the life of the tab but it seems to be the life of the browser instead.

Yes, a bit more reading makes me realize this!

>  I think it makes more sense (and drastically simplifies the code) for this to just be per page.

Oh! Um, that means I could just store these values in an `array`. And everytime the user refreshes the page, all of this starts afresh? 

> We should also make sure to only display the message to the user once. We don't want them to keep seeing it.

Yes, we do not want to annoy them!!